### PR TITLE
Initial commit for Get-DbaDatabaseState

### PR DESCRIPTION
**Pull request by niphlod**
How to test this code: 
- [ ] database state consistent with the one in the backend (combinations included, i.e. "offline\read-only")
- [ ] no "zero-width" properties are returned (i.e. no '' on RW, Status, Access)

Has been tested on minimum requirements:
- [ ]  Powershell 3
- [ ]  Windows 7
- [ ]  SQL Server 2000

Has been tested on maximum requirements:
- [ ]  SQL Server vNext
- [x]  Windows 10
- [ ]  Azure Database

Tests for tester:
- [ ] Working/useful help content, including link to command on dbatools web site
- [ ] All examples work as advertised
- [ ] Does not contain template content
- [ ] Does not contain excessive/unnecessary amounts of comments
- [ ] Works remotely
- [ ] Works locally
- [ ] Works on lower versions or throws error specifying version not supported
- [ ] Works with named instances
- [ ] Works with clustered instances
- [ ] Handles offline/read only databases
- [ ] Supports multiple servers (at the command line or piped from Get-SqlRegisteredServerName)
- [ ] No un-handled errors which stop the command working with multiple servers

### CODE REVIEW

File: functions/Get-DbaDatabaseState.ps1 [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+.PARAMETER Credential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER Database
+Gets options only on these databases
+
+.PARAMETER Exclude
+Gets options for all but these specific databases
+
+.PARAMETER WhatIf
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseState.ps1]: whatif and confirm help will be autogenerated if shouldprocess = $true.

File: functions/Get-DbaDatabaseState.ps1 [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+Gets options for all databases of the sqlserver2014a instance except HR
+
+.EXAMPLE
+Get-DbaDatabaseState -SqlServer sqlserver2014a, sqlserver2014b
+
+Gets options for all databases of sqlserver2014a and sqlserver2014b instances
+
+#>
+	[CmdletBinding(SupportsShouldProcess = $true)]
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseState.ps1]: supportshouldprocess would not be necessary for Get- functions.

File: functions/Get-DbaDatabaseState.ps1 [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+				if($db.Status -match $status)
+				{
+					$base.Status = $StatusHash[$status]
+					break
+				}
+			}
+			return $base
+		}
+
+		$RWExclusive = @('ReadOnly', 'ReadWrite')
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseState.ps1]: Why are those 3 variables here? I don't see them being used.

File: functions/Get-DbaDatabaseState.ps1 [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+.EXAMPLE
+Get-DbaDatabaseState -SqlServer sqlserver2014a -Exclude HR
+
+Gets options for all databases of the sqlserver2014a instance except HR
+
+.EXAMPLE
+Get-DbaDatabaseState -SqlServer sqlserver2014a, sqlserver2014b
+
+Gets options for all databases of sqlserver2014a and sqlserver2014b instances
+
```
Comment (PowerDBAKlaas) [inline functions/Get-DbaDatabaseState.ps1]: Please include an example
'sql1','sql2' | Get-DbaDatabaseState

File: functions/Get-DbaDatabaseState.ps1 (hunk 1 above)
Comment (niphlod) [inline functions/Get-DbaDatabaseState.ps1]: didn't know. to be fair, I copy-pasted from existing commands. Will fix it.

**Comment by PowerDBAKlaas** [comment]
Great, thanks

**Comment by ctrlbold** [comment]
This was awesome! Worked well across all of my instances.

**Pull request merged**